docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
+* `sphinx.ext.intersphinx
+  <https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html>`_
+  for linking to other projects' documentation
+* `sphinx.ext.graphviz
+  <https://www.sphinx-doc.org/en/master/usage/extensions/graphviz.html>`_
+  to allow `Graphviz <https://graphviz.org/>`_ graphs to be included
+* `sphinx.ext.mathjax
+  <https://www.sphinx-doc.org/en/master/usage/extensions/math.html#module-sphinx.ext.mathjax>`_
+  for math rendering with `MathJax <https://www.mathjax.org/>`_
+* `sphinx.ext.napoleon
+  <https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html>`_
+  for allowing NumPy style docstrings
+* `sphinx.ext.todo
+  <https://www.sphinx-doc.org/en/master/usage/extensions/todo.html>`_ to support
+  ``todo`` directives
+* `nbsphinx <https://nbsphinx.readthedocs.io>`_ for including
+  `Jupyter`_ notebooks
+* `sphinx_copybutton <https://sphinx-copybutton.readthedocs.io>`_ to add
+  a "copy" button for code blocks
+* `sphinx_gallery.load_style
+  <https://sphinx-gallery.github.io/stable/advanced.html?highlight=load_style#using-only-sphinx-gallery-styles>`_
+  for using sphinx-gallery styles
+* IPython.sphinxext.ipython_console_highlighting
+* `sphinx_changelog <https://sphinx-changelog.readthedocs.io>`_
+  for rendering `towncrier`_ changelogs
+* `plasmapy_sphinx` for customizations created for use in PlasmaPy.
+  Note that `plasmapy_sphinx` is expected to be broken out into its own
+  package at a future date.
+
+These extensions are specified in ``extensions`` in `docs/conf.py`_.
+
+References to other packages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Intersphinx_ allows the automatic generation of links to the
+documentation of objects in other projects. The mappings are defined in
+the ``intersphinx_mapping`` dictionary in `docs/conf.py`_, and include
+`python`, `numpy`, `scipy`, `astropy`, `pandas`, `sphinx`, and
+`sphinx_automodapi`.
+
+When we include ``astropy.units.Quantity`` in the documentation, it
+will show up as `astropy.units.Quantity` with a link to Astropy's
+documentation.
+
+Substitutions
+~~~~~~~~~~~~~
+
+Some functions and classes are referred to repeatedly throughout the
+documentation. reST allows us to `define substitutions
+<https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#substitution-definitions>`_.
+
+.. code-block:: rst
+
+   .. |Particle| replace:: `~plasmapy.particles.particle_class.Particle`
+
+PlasmaPy has certain common substitutions pre-defined so that they can
+be used elsewhere in the documentation. For example, we can write
+``|Quantity|`` instead of ``~astropy.units.Quantity``, and
+``|Particle|`` instead of ``~plasmapy.particles.particle_class.Particle``.
+For an up-to-date list of substitutions, please refer to the
+`docs/common_links.rst`_ file.
+
+Because substitutions are performed when `sphinx` builds the
+documentation, they will not be performed before `help` accesses the
+docstring of an `object`. For example, when ``|Particle|`` is used in
```

Comment:
```suggestion
Since substitutions are performed by `sphinx` when the documentation
is built, any substitution used in docstrings will not show up when using
Python's `help` function (or the like). For example, when ``|Particle|`` is used in
```